### ISSUE TITLE
Resumable downloads

### ISSUE THREAD

**Issue by mgallien** [nextcloud/desktop#4966]
When downloading a big file, if the transfer gets interrupted, it would be nice to resume the download from the already downloaded data instead of starting from scratch.

**Comment by mgallien** [comment]
@tobiasKaminsky

**Comment by PhilippSchlesinger** [comment]
the same holds true for uploads :-)

**Comment by tobiasKaminsky** [comment]
On uploads we already have this, right @mgallien?

**Comment by mgallien** [comment]
> On uploads we already have this, right @mgallien?

I do not think so
According to https://httpwg.org/specs/rfc9110.html#field.range I do not think there is a way to resume an upload

**Comment by mgallien** [comment]
> the same holds true for uploads :-)

we would already resume properly chunked uploads
for smaller one, I do not know how that would be possible

**Comment by tobiasKaminsky** [comment]
Right, I meant chunked upload. Small uploads <10 MB can be restarted, as costs for this can be ignored…

**Comment by PhilippSchlesinger** [comment]
> > On uploads we already have this, right @mgallien?
> 
> I do not think so According to https://httpwg.org/specs/rfc9110.html#field.range I do not think there is a way to resume an upload

maybe this could work via the PATCH method according to 
https://httpwg.org/specs/rfc5789.html

**Comment by PhilippSchlesinger** [comment]
> Right, I meant chunked upload. Small uploads <10 MB can be restarted, as costs for this can be ignored…

Ah, great, resume is already implemented for chunked uploads larger than 10 MB!
And yes, for small files this can be neglected.

**Comment by pedlino** [comment]
+1
Downloads are not resumable from browser and there is no option to copy a valid direct link to use in a download manager

**Comment by nfnfgo** [comment]
Is there any update on this lol, it will be good to be able to resume the download where it stopped last time. Especially when the files are large (several GB per file or even larger)

**Comment by AinzRimuru** [comment]
> Is there any update on this lol, it will be good to be able to resume the download at where it stopped last time. Especially when the files are large (several GB per file or even larger)
> 
> I'm using the latest version of the NextCloud Desktop, and it seems that NextCloud will create a cache when syncing(downloading) large files, and maybe those cache could be used to support such feature?

There is no update, and during the process of downloading the file, I found that the temporary cache gets deleted when the download is interrupted and the download starts over again. The cached part is not being effectively utilized.

**Comment by Rello** [comment]
already implemented

**Comment by nfnfgo** [comment]
Sorry for replying to a closed issue, however, it seems that the download is still not resumable with latest Windows NextCloud Desktop client.

Concretely, the procedure is:

1. Set a large file as "Always available offline"
2. NextCloud start downloading it
3. Check the target location of the downloading file, we could see a tmp file created by NextCloud
4. Interrupt the download (Turn off the Wi-Fi / close the TCP connection manually / Stop NextCloud client)
5. Reconnect to the Internet, resume the download
6. Check the download location again, the old tmp file has been cleared and new tmp file created, and  download start from the very beginning of the file

**Comment by mortie** [comment]
I don't think this is implemented? In order for a browser to be able to resume a download, the server needs to respond with `Accept-Ranges: bytes` (and then actually support range requests of course). When I inspect responses using `curl -I <download URL>`, I see no `Accept-Ranges` header.

Can this issue be re-opened?
